test/node-api/test_reference_by_node_api_version/test.js

```diff
@@ -0,0 +1,124 @@
+'use strict';
+// Flags: --expose-gc
+//
+// Testing API calls for Node-API references.
+// We compare their behavior between Node-API version 8 and later.
+// In version 8 references can be created only for object, function,
+// and symbol types, while in newer versions they can be created for
+// any value type.
+//
+const { gcUntil, buildType } = require('../../common');
+const assert = require('assert');
+const addon_v8 = require(`./build/${buildType}/test_reference_obj_only`);
+const addon_new = require(`./build/${buildType}/test_reference_all_types`);
+
+async function runTests(addon, isVersion8, isLocalSymbol) {
+  let allEntries = [];
+
+  (() => {
+    // Create values of all napi_valuetype types.
+    const undefinedValue = undefined;
+    const nullValue = null;
+    const booleanValue = false;
+    const numberValue = 42;
+    const stringValue = 'test_string';
+    const globalSymbolValue = Symbol.for('test_symbol_global');
```

Comment:
Do we still need this distinction between global and local symbols?

Reply:
Yes, we need it because they behave differently when the `napi_ref` ref count becomes 0 and we run GC.
My previous test did not make a distinction between the two types because I only used a global symbol.
It caused me wrongly to believe that Symbols are never weak. 
It is good to show/test the different behavior for these two types of symbols.